### PR TITLE
Skip blacklisted app translations in downloads

### PR DESCRIPTION
**Pull request by kaapstorm**
Small spec: https://docs.google.com/document/d/1sN1XOlSMqYKDgj-n7sK1VAjkZijmtqUOhxE9vcC1vXg/

TLDR: Projects can "blacklist" translations so they are left out of data sent to Transifex. This PR allows blacklisted translations to be left out of app translation downloads too.

@orangejenny cc buddy @calellowitz

### CODE REVIEW

**Comment by kaapstorm** [comment]
Feature flag: Extends functionality currently available behind "**Translate Application Content With Transifex**"

File: corehq/apps/app_manager/templates/app_manager/partials/bulk_upload_app_translations.html
Comment (mkangia) [inline corehq/apps/app_manager/templates/app_manager/partials/bulk_upload_app_translations.html]: My understanding is that this should exclude
1. blacklisted translations
2. translations that have "Skipped transifex" set on them, which includes labels, forms or complete modules. Check out `AppTranslationsGenerator`

Currently this PR is addressing just the first one but needs to skip both

If that is true, this sentence can be updated to say
"Exclude translations not sent for transifex" or
"Download translations eligible for transifex only"

Blacklisted translation was the name used for same purpose, so i think just saying that should be fine.

**Comment by mkangia** [comment]
fyi @emord

**Comment by kaapstorm** [comment]
Probably best to :tropical_fish:

File: corehq/apps/translations/generators.py
Comment (kaapstorm) [inline corehq/apps/translations/generators.py]: @emord does this get used anywhere?

Comment (emord) [reply corehq/apps/translations/generators.py]: I think here: https://github.com/dimagi/commcare-hq/blob/4119e3b6b390ae3eca91fa7d89564296596e5cee/corehq/apps/translations/integrations/transifex/parser.py#L35 ?

Comment (kaapstorm) [reply corehq/apps/translations/generators.py]: :+1: Thanks. That looks right.

File: corehq/apps/app_manager/templates/app_manager/partials/bulk_upload_app_translations.html
Comment (mkangia) [inline corehq/apps/app_manager/templates/app_manager/partials/bulk_upload_app_translations.html]: nit: do we need the '?' in the end?

File: corehq/apps/translations/app_translations/download.py
Comment (mkangia) [inline corehq/apps/translations/app_translations/download.py]: add the other place where we do something similar,
https://github.com/dimagi/commcare-hq/pull/24604/files#diff-27f0b75dc11241833e40eb967bbb737aR91
we are assuming more than 3 entries in `module_row`, fine to ignore that here?

File: corehq/apps/translations/app_translations/download.py
Comment (mkangia) [inline corehq/apps/translations/app_translations/download.py]: this block should be removed now?

File: corehq/apps/app_manager/static/app_manager/js/settings/translations.js
Comment (mkangia) [inline corehq/apps/app_manager/static/app_manager/js/settings/translations.js]: since the approach has changed to a broader translations and not just blacklisted, the naming would need to adapt to that.
How about `onlyTransifexEligible`?
Similarly in the url, `skipbl` would need to change.

File: corehq/apps/translations/app_translations/download.py
Comment (mkangia) [inline corehq/apps/translations/app_translations/download.py]: can extract a private method for better readability
```
def include_module_or_form(module_or_form):
    return eligible_for_transifex_only and checker.exclude_module(module)
```
and then use that here and for the form.

^ the logic above is not accurate but just an idea

Comment (mkangia) [reply corehq/apps/translations/app_translations/download.py]: Similarly for module_row. You can also consider filter all module_rows and then just appending them all at once.

Comment (mkangia) [reply corehq/apps/translations/app_translations/download.py]: May be for form_media as well. Basically extracting out the blocks of logic into methods so that the naming can help understand them quickly and the logic is contained in a method for better absorption of what all is happening in this long method.

File: corehq/apps/translations/generators.py
Comment (mkangia) [inline corehq/apps/translations/generators.py]: This is a great addition.
how about moving it to a file of its own and under the 'integrations.transifex' since it relates that.
so then this would become
`corehq.apps.translations.integrations.transifex.checker.EligibleForTransifex`, also dropping the Checker in the end.

**Review by mkangia — APPROVED** [review]
nothing blocking. This must be tested before merging. I think in the past, Jordan lerner has tested it from GS but QA has not.

**Comment by kaapstorm** [comment]
Cool. I will deploy to staging for Jordan Lerner. They seem super keen for this to be available as soon as possible, so I will probably follow up PR feedback in a second PR if Jordan OKs this before I get to it.

File: corehq/apps/translations/app_translations/download.py
Comment (kaapstorm) [reply corehq/apps/translations/app_translations/download.py]: There are only ever three values in `module_row` for single-sheet uploads because there is only one language.

File: corehq/apps/app_manager/templates/app_manager/partials/bulk_upload_app_translations.html
Comment (kaapstorm) [reply corehq/apps/app_manager/templates/app_manager/partials/bulk_upload_app_translations.html]: Probably pedantic, but when a checkbox is used to select options, then there wouldn't be a question mark because the UI is telling you to choose something. But when you are using a checkbox to answer a question (essentially "Do you want to download only translations eligible for Transifex?" "Yes" or "No") then the question should end in a question mark. I think if I were to change this, it would be to keep the question mark, but replace the checkbox with a yes-no toggle/switch.

File: corehq/apps/translations/app_translations/download.py
Comment (kaapstorm) [reply corehq/apps/translations/app_translations/download.py]: (I love how Github lets you do the edits you missed :) )

**Comment by kaapstorm** [comment]
Responded to [QA feedback](https://app.asana.com/0/968130631790203/1122969855795967/f). Dropping "Do not merge" label.

**Pull request merged**
